collect-assets.js

```diff
@@ -0,0 +1,117 @@
+import {
+  existsSync,
+  mkdirSync,
+  createWriteStream,
+  createReadStream,
+  writeFileSync,
+  readdirSync,
+  renameSync,
+} from "fs";
+import { resolve as _resolve, join, dirname } from "path";
+import { fileURLToPath } from "url";
+import fetch from "node-fetch";
+import { Extract } from "unzipper";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const GITHUB_API = "https://api.github.com";
+const REPOSITORY = "konveyor/kai";
+const RELEASE_TAG = "v0.0.1";
+const DOWNLOAD_DIR = _resolve(__dirname, "downloaded_assets");
+const META_FILE = _resolve(DOWNLOAD_DIR, "collect.json");
+
+const ASSETS_TO_DOWNLOAD = [
+  { name: "kai-rpc-server.linux-x86_64.zip", folder: "linux", platform: "linux" },
+  { name: "kai-rpc-server.macos-arm64.zip", folder: "macos-arm", platform: "macos-arm" },
+  { name: "kai-rpc-server.macos-x86_64.zip", folder: "macos-x86", platform: "macos-x86" },
+  { name: "kai-rpc-server.windows-x86_64.zip", folder: "windows", platform: "windows" },
+];
+
+if (!existsSync(DOWNLOAD_DIR)) {
+  mkdirSync(DOWNLOAD_DIR, { recursive: true });
+}
+
+async function getReleaseMetadata() {
+  const url = `${GITHUB_API}/repos/${REPOSITORY}/releases/tags/${RELEASE_TAG}`;
+  console.log(`Fetching release metadata for tag: ${RELEASE_TAG}`);
+
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`Failed to fetch release metadata: ${response.statusText}`);
+  }
+
+  return await response.json();
+}
+
+async function downloadAndExtractAsset(asset, folder, platform) {
+  const platformDir = join(DOWNLOAD_DIR, folder);
+  const assetPath = join(platformDir, asset.name);
+
+  if (!existsSync(platformDir)) {
+    mkdirSync(platformDir, { recursive: true });
+  }
+
+  console.log(`Downloading asset: ${asset.name}`);
+  const response = await fetch(asset.browser_download_url);
+  if (!response.ok) {
+    throw new Error(`Failed to download ${asset.name}: ${response.statusText}`);
+  }
+
+  const fileStream = createWriteStream(assetPath);
+  await new Promise((resolve, reject) => {
+    response.body.pipe(fileStream);
+    response.body.on("error", reject);
+    fileStream.on("finish", resolve);
```

Comment:
Using node's `fetch()`...
```suggestion
    const reader = Readable.fromWeb(response.body);
    reader.pipe(fileStream);
    reader.on("error", reject);
    fileStream.on("finish", resolve);
```